packages/plugin-ext/src/plugin/preference-registry.ts

```diff
@@ -238,7 +238,7 @@ export class PreferenceRegistryExtImpl implements PreferenceRegistryExt {
     private readonly OVERRIDE_PROPERTY_PATTERN = new RegExp(this.OVERRIDE_PROPERTY);
 
     private parseConfigurationData(data: { [key: string]: any }): { [key: string]: any } {
-        return Object.keys(data).reduce((result: any, key: string) => {
+        return Object.keys(data).filter(key => !key.startsWith('__proto__.')).reduce((result: any, key: string) => {
```

Comment:
Let's make the filter such as:

```ts
filter(key => /(^|\.)__proto__\b/.test(key))
```

Else the following key could trigger an injection: `a.__proto__.injected = true`.

Reply:
I added a new testcase but your filter was not working.
I used `includes` now

Reply:
I forgot to negate it 🤦 I updated it. I would also insist on using the regex because it will prevent people from overriding the `__proto__` property alone of the object we use. This doesn't prevent prototype pollution, but it prevents overriding an object's prototype for no good reason. e.g. `'__proto__': {}` will replace the object prototype with an empty object.

I pushed what I proposed + a new test case, feel free to either remove it or integrate it in your own commit.

Reply:
I'm fine with your proposal

Reply:
I'll need an approval as well :-)